core/src/main/java/hudson/cli/DeleteCommand.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2015 Red Hat, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package hudson.cli;
+
+import hudson.AbortException;
+import jenkins.model.Jenkins;
+
+import java.util.List;
+import java.util.HashSet;
+
+/**
+ * CLI command, which helps delete job(s) or node(s).
+ * @since TODO
+ */
+
+public abstract class DeleteCommand<T> extends CLICommand {
+
+    @SuppressWarnings("MismatchedQueryAndUpdateOfCollection")
+
+    protected abstract void tryDelete(String name, Jenkins jenkins) throws Exception;
+
+    protected void checkExists(T item, String item_s, String item_type) throws Exception{
+        if(item == null) {
+            throw new IllegalArgumentException("No such " + item_type + " '" + item_s + "'");
+        }
+        return;
+    }
+
+    protected int deleteItems(List<String> items) throws Exception {
+
+        boolean errorOccurred = false;
+        final Jenkins jenkins = Jenkins.get();
+
+        final HashSet<String> hs = new HashSet<>(items);
+
+        for (String item_s: hs) {
```

Comment:
Please use more descriptive variable names that are more in line with the existing codebase.

This looks similar to Hungarian notation, which we're not using. If it's the name of something, call it `somethingName`, rather than `something_s`.

Plus this still uses the "item" term which we've determined is just confusing.